vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.cc

```diff
@@ -1644,4 +1649,161 @@ void LedgerImpl::FetchBalance(ledger::FetchBalanceCallback callback) {
   bat_wallet_->FetchBalance(callback);
 }
 
+void LedgerImpl::OnGetAllTransactions(
```

Comment:
trying to put all these different types together into a single query seems really overly complicated to me. I think we either need a new type with its own database query or do multiple queries to Ledger to get the different data types

Reply:
These have been split out.